postgres/datadog_checks/postgres/version_utils.py

```diff
@@ -15,58 +17,76 @@
 V10 = VersionInfo(**semver.parse("10.0.0"))
 
 
-def get_raw_version(db):
-    cursor = db.cursor()
-    cursor.execute('SHOW SERVER_VERSION;')
-    raw_version = cursor.fetchone()[0]
-    return raw_version
-
-
-def is_aurora(db):
-    cursor = db.cursor()
-    try:
-        cursor.execute("select exists(select 1 from pg_proc where proname = 'aurora_version');")
-        return cursor.fetchone()[0]
-    except Exception:
-        db.rollback()
-        return False
+class VersionUtils(object):
+    def __init__(self):
+        self.log = get_check_logger()
 
+    @staticmethod
+    def get_raw_version(db):
+        cursor = db.cursor()
+        cursor.execute('SHOW SERVER_VERSION;')
+        raw_version = cursor.fetchone()[0]
+        return raw_version
 
-def parse_version(raw_version):
-    try:
-        # Only works for MAJOR.MINOR.PATCH(-PRE_RELEASE)
-        return semver.parse_version_info(raw_version)
-    except ValueError:
-        pass
-    try:
-        # Version may be missing minor eg: 10.0
-        version = raw_version.split(' ')[0].split('.')
-        version = [int(part) for part in version]
-        while len(version) < 3:
-            version.append(0)
-        return VersionInfo(*version)
-    except ValueError:
-        # Postgres might be in development, with format \d+[beta|rc]\d+
-        match = re.match(r'(\d+)([a-zA-Z]+)(\d+)', raw_version)
-        if match:
-            version = list(match.groups())
-            return semver.parse_version_info('{}.0.0-{}.{}'.format(*version))
-    raise Exception("Cannot determine which version is {}".format(raw_version))
+    def is_aurora(self, db):
+        cursor = db.cursor()
+        try:
+            # This query is preferred to the one below cause it does not pollute PG logs with errors
+            cursor.execute("select exists(select 1 from pg_proc where proname = 'aurora_version');")
+            result = cursor.fetchone()[0]
+            if result is not None:
+                # In some Aurora versions the above query returns None
```

Comment:
Do we have more details about that? If so can we document it here?

Reply:
Not really, we're logging the result in https://github.com/DataDog/integrations-core/blob/master/postgres/datadog_checks/postgres/postgres.py#L442 and it was showing as None

Reply:
Oh but we're using `self._is_aurora` here, which can be None if `self.is_aurora` has not been called yet.

Reply:
It was showing as None multiple times, the reason is we were setting it to None when the exception was captured